nservicebus/upgrades/nhibernate-timeouts-index.md

```diff
@@ -0,0 +1,111 @@
+---
+title: NHibernate Persistence - Resolving incorrect timeout table schema (#252)
+summary: Instructions on how to resolve incorrect schema that can cause performance issues for affected versions 6 to 7.
+component: NHibernate
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 5
+ - 6
+---
+
+
+## Summary
+
+This guidance explains how to resolve an incorrectly created schema when passing a custom NHibernate configuration to NServiceBus as described in the following issue:
+
+- https://github.com/Particular/NServiceBus.NHibernate/issues/252
+
+This issues will cause performance issues if the table contains a large number of rows. Inserts and queries are inefficient due to the incorrect order of columns. This results in unnecessary locking which limits the processing throughput of timeouts.
+
+
+## Compatibility
+
+This issues has been resolved in following patch versions as defined in our  [support policy](support-policy.md):
+
+- 7.1.6 in use by NServiceBus 6.x
+- 7.0.6 (support 7.0.x) in use by NServiceBus 5.x
+- 6.2.8 (support 6.2.x) in use by NServiceBus 5.x
+
+
+
+If you are using any of the supported minor versions (7.1.x, 7.0.x, or 6.2.8) then you should at least update to the latest patch release. If you are using an older version then you should update to a newer minor (in case of 6.1.x or 6.0.x) or major version (any version prior to 6.x).
+
+
+## Upgrade steps
+
+Steps:
+
+ * Update to latest patch release
+ * Deploy the new version
+ * Check if you get a warning related to this schema issue
+   * Or manually inspect the schema in your database
+ * Follow procedure on how to resolve schema issues for your database engine (Microsoft SQL Server or Oracle)
+   * If you are using any other database engine then you have to apply these changes in the database dialact for your engine manually
+
+
+## Check at startup
+
+If you have endpoints with an incorrect table schema then this is detected in all fixed supported versions (at March 1th, 2017) for 6.2.x, 7.0.x and 7.1.x. The detection routine is run when the endpoint instance is created. If you are affected you will get the following log event with log level warning:
+
+> Could not find TimeoutEntity_EndpointIdx index. This may cause significant performance degradation of message deferral. Consult NServiceBus NHibernate persistence documentation for details on how to create this index.
+
+If this log event is written to your log file then read the following guidance on how to apply corrections.
+
+
+## Potential issues
+
+Any of the following issues can be present:
+
+- table `TimeoutEntity` has a clustered primary key
```

Comment:
What do you mean? Doesn't Oracle have clustered indexes or is the PK always clustered with Oracle or does it use a different name?